packages/ui-core/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,191 @@
+import debounce from 'lodash.debounce';
+import './Tooltip.less';
+
+const IS_TOUCH_DEVICE = detectTouch();
+
+export const Placement = {
+    LEFT: 'left',
+    TOP: 'top',
+    RIGHT: 'right',
+    BOTTOM: 'bottom',
+} as const;
+
+type PlacementType = typeof Placement[keyof typeof Placement];
+
+export const Size = {
+    SMALL: 'small',
+    MEDIUM: 'medium',
+ } as const;
+
+type SizeType = typeof Size[keyof typeof Size];
+
+export const Trigger = {
+    HOVER: 'hover',
+    CLICK: 'click',
+ } as const;
+
+type TriggerType = typeof Trigger[keyof typeof Trigger];
+
+type IsOpenType = boolean;
+
+export interface ITooltipProps {
+    /** Tooltip position */
+    placement?: PlacementType;
+    /** Size of padding in tooltip content */
+    size?: SizeType;
+    /** Trigger handler type */
+    trigger?: TriggerType;
+    /** Trigger element */
+    triggerElement: Element;
+    /** Manipulate open state from outside */
+    isOpened?: IsOpenType;
+    /** Custom class name */
+    className?: string;
+    /** Content */
+    children?: React.ReactNode;
+    /** Open handler */
+    onOpen?: (e: MouseEvent) => void;
+    /** Close handler */
+    onClose?: (e: MouseEvent) => void;
+}
+
+const cn = cnCreate('mfui-beta-tooltip');
+const Tooltip: React.FC<ITooltipProps>  = ({
+    className,
+    placement = Placement.TOP,
+    size = Size.MEDIUM,
+    trigger = Trigger.HOVER,
+    triggerElement,
+    isOpened = false,
+    children,
+    onOpen,
+    onClose,
+}) => {
+    const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
+    const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);
+    const options = {
+        placement,
+        modifiers: [
+            {
+                name: 'arrow',
+                options: { element: arrowElement },
+            },
+            {
+                name: 'flip',
+                options: {
+                    fallbackPlacements: ['left', 'right' , 'top' , 'bottom'],
+                },
+            },
+        ],
+    };
+    const { styles, attributes } = usePopper(triggerElement, popperElement, options);
+
+    const [isOpen, setIsOpen] = useState<IsOpenType>(isOpened);
+    useEffect(() => setIsOpen(isOpened), [isOpened]);
+
+    const closeTooltip = useCallback((e: MouseEvent): void => {
+        setIsOpen(false);
+        onOpen && onOpen(e);
+    }, [isOpen, onClose]);
+
+    const handleOpenEvent = useCallback((e: MouseEvent): void => {
+        if (!isOpen) {
+            setIsOpen(true);
+            onOpen && onOpen(e);
+        }
+    }, [isOpen, onOpen]);
+
+    const handleClickOutsideEvent = (e: MouseEvent): void => {
+        if (e.target instanceof Node && popperElement && !popperElement.contains(e.target)) {
+            closeTooltip(e);
+        }
+
+        return;
+    };
+
+    const handleMouseOverOutsideEvent = (e: MouseEvent): void => {
+        if (
+            e.target instanceof Node &&
```

Comment:
Условные выражения на три строки лучше разбить на несколько переменных.
```suggestion
            const isElement = e.target instanceof Element;
            const isOnPopperElement = popperElement && popperElement.contains(e.target);
            const isOnTriggerElement = triggerElement && triggerElement.contains(e.target);
            if (isElement && !isOnPopperElement && !isOnTriggerElement)
```
С логикой мог ошибиться, но примерно так. В предыдущем обработчике тоже можно две переменные создать.